edx_rest_framework_extensions/utils.py

```diff
@@ -38,21 +52,31 @@ def jwt_decode_handler(token):
 
     options = {
         'verify_exp': api_settings.JWT_VERIFY_EXPIRATION,
+        'verify_aud': settings.JWT_AUTH['JWT_VERIFY_AUDIENCE'],
         'require_exp': True,
-        'require_iat': True,
+        'require_iat': True
     }
 
-    try:
-        return jwt.decode(
-            token,
-            api_settings.JWT_SECRET_KEY,
-            api_settings.JWT_VERIFY,
-            options=options,
-            leeway=api_settings.JWT_LEEWAY,
-            audience=api_settings.JWT_AUDIENCE,
-            issuer=api_settings.JWT_ISSUER,
-            algorithms=[api_settings.JWT_ALGORITHM]
-        )
-    except jwt.InvalidTokenError:
-        logger.exception('JWT decode failed!')
-        raise
+    # JWT_ISSUERS is not one of DRF-JWT's default settings, and cannot be accessed
+    # using the `api_settings` object without overriding DRF-JWT's defaults.
+    for jwt_issuer in get_jwt_issuers():
+        try:
+            decoded = jwt.decode(
+                token,
+                jwt_issuer['SECRET_KEY'],
+                api_settings.JWT_VERIFY,
+                options=options,
+                leeway=api_settings.JWT_LEEWAY,
+                audience=api_settings.JWT_AUDIENCE,
+                issuer=jwt_issuer['ISSUER'],
+                algorithms=[api_settings.JWT_ALGORITHM]
+            )
+            return decoded
+        # Ignore decoding errors since we have multiple issuers to check jwt.InvalidTokenError
+        except jwt.DecodeError:
+            msg = "Token decode failed for issuer '{issuer}'".format(issuer=jwt_issuer['ISSUER'])
+            logger.exception(msg)
+
+    msg = 'All combinations of JWT issuers and secret keys failed to validate the token.'
+    logger.exception(msg)
```

Comment:
Does this work outside of an `except` block? It should probably be `logger.error`.